lib/server/handlers/sthRemoveDataHandler.js

```diff
@@ -64,7 +64,13 @@ function removeDataHandler(request, reply) {
         },
         function(err) {
             if (err) {
-                if (err.name === 'MongoError' && err.message.indexOf('does not exist. Currently in strict mode')) {
+                if (err.name === 'Not connected') {
+                    sthLogger.error(request.sth.context, 'Database is not connected');
+                    response = reply(err, { statusCode: 500 }).code(500);
```

Comment:
This change would make the test pass. In this case, the response when making a delete request without the DB connection will not be a 500 code. 
```suggestion
                    response = reply(err, { statusCode: 500 });
```